pkg/konnector/konnector_reconcile.go

```diff
@@ -120,6 +120,9 @@ func (r *reconciler) reconcile(ctx context.Context, binding *kubebindv1alpha1.AP
 	ctrl, err := r.newClusterController(
 		binding.Spec.KubeconfigSecretRef.Namespace+"/"+binding.Spec.KubeconfigSecretRef.Name,
 		providerNamespace,
+		func(svcBinding *kubebindv1alpha1.APIServiceBinding) bool {
+			return svcBinding.Name == binding.Name
```

Comment:
nearly. You have to check that `svcBinding.Name` is in `ctrlContext.serviceBindings`, while taking the `r.lock`. Maybe it's a good idea to turn the lock into a RWMutex.